src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs

```diff
@@ -412,11 +415,12 @@ private IEnumerable<AnalyzerReference> GetAnalyzerReferences(PackageDescription
             var analyzers = package
                 .PackageLibrary
                 .Files
-                .Where(path => path.StartsWith("analyzers" + Path.DirectorySeparatorChar) &&
+                .Where(path => path.StartsWith("analyzers" + LockFile.DirectorySeparatorChar) &&
```

Comment:
Try changing this back to the current platform and see if your analyzer bug goes away.